electron-app/src/server/websites/misskey/misskey.service.ts

```diff
@@ -0,0 +1,406 @@
+    data: MissKeyAccountData,
+    file: PostFile,
+    altText: string,
+  ): Promise<{ id: string }> {
+
+    const M = generator('misskey', `https://${data.website}`, data.tokenData.access_token);
+    const upload = await M.uploadMedia(file.value, { description: altText });
+    console.log(upload)
+    if (upload.status > 300) {
+      return Promise.reject(
+        this.createPostResponse({ additionalInfo: upload.status, message: upload.statusText }),
+      );
+    }
+
+    return { id: upload.data.id };
+  }
+
+  async postFileSubmission(
+    cancellationToken: CancellationToken,
+    data: FilePostData<MissKeyFileOptions>,
+    accountData: MissKeyAccountData,
+  ): Promise<PostResponse> {
+    const M = generator('misskey', `https://${accountData.website}`, accountData.tokenData.access_token);
+
+    const files = [data.primary, ...data.additional];
+    this.checkCancelled(cancellationToken);
+    const uploadedMedias: {
+      id: string;
+    }[] = [];
+    for (const file of files) {
+      uploadedMedias.push(await this.uploadMedia(accountData, file.file, data.options.altText));
+    }
+
+    const instanceInfo: MissKeyInstanceInfo = this.getAccountInfo(data.part.accountId, INFO_KEY);
+    const chunkCount = instanceInfo ? instanceInfo?.configuration?.statuses?.max_media_attachments : 4;
+    const maxChars = instanceInfo ? instanceInfo?.configuration?.statuses?.max_characters : 500;
+
+    const isSensitive = data.rating !== SubmissionRating.GENERAL;
+    const { options } = data;
+    const chunks = _.chunk(uploadedMedias, chunkCount);
+    let lastId = undefined;
+
+    for (let i = 0; i < chunks.length; i++) {
+      let statusOptions: any = {
+        status: '',
+        sensitive: isSensitive,
+        visibility: options.visibility || 'public',
+        spoiler_text: ''
+      };
+      let status = undefined;
+
+      let form = undefined;
+      if (i === 0) {
+        status = `${options.useTitle && data.title ? `${data.title}\n` : ''}${
+          data.description
+        }`.substring(0, maxChars)
+
+        statusOptions = {
+          sensitive: isSensitive,
+          visibility: options.visibility || 'public',
+          media_ids: chunks[i].map((media) => media.id), 
+          spoiler_text: "",
+        }
+     } else {
+      statusOptions = {
+        sensitive: isSensitive,
+        visibility: options.visibility || 'public',
+        media_ids: chunks[i].map((media) => media.id),
+        in_reply_to_id: lastId,  
+        spoiler_text: "",    
+      }      
+     }
+
+      const tags: string[] = this.parseTags(data.tags, {
+        spaceReplacer: '_',
+        minLength: 1,
+        maxLength: 100,
+      });
+
+      // Update the post content with the Tags if any are specified - for MissKey, we need to append 
+      // these onto the post, *IF* there is character count available.
+      if (tags.length > 0) {
+        status += "\n\n";
+      }
+
+      tags.forEach(tag => {
```

Comment:
Same comment here as in mastodon service.

Reply:
Resolved